File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num < 2) return true;
        long left = 2 , right = num / 2;
        long x , guessNum;
        while(left <= right){
            x = left + (right - left) / 2;
            guessNum = x * x;
            if(guessNum == num) return true;
            else if(guessNum < num)left = x +1;
            else{
                right = x - 1;
            }
        }
        return false;
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        if(num < 2) return true;
        long guess = 0 , mid ;
        long left = 2 , right = num / 2 ;

        while(left <= right ){
             mid = left + (right - left ) / 2 ;
            guess = mid * mid;
            if(num == guess) return true;
            if(guess < num) left = mid +1 ;
            else right = mid - 1;
        }
        return false;
    }
}